src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -158,21 +232,21 @@ private BoundExpression BindDeconstructionAssignment(AssignmentExpressionSyntax
         private class DeconstructionVariable
         {
             public readonly BoundExpression Single;
-            public readonly ImmutableArray<DeconstructionVariable> Nested;
+            public readonly ArrayBuilder<DeconstructionVariable> Nested;
 
             public DeconstructionVariable(BoundExpression variable)
             {
                 Single = variable;
-                Nested = default(ImmutableArray<DeconstructionVariable>);
+                Nested = null;
             }
 
-            public DeconstructionVariable(ImmutableArray<DeconstructionVariable> variables)
+            public DeconstructionVariable(ArrayBuilder<DeconstructionVariable> variables)
             {
                 Single = null;
                 Nested = variables;
             }
 
-            public bool IsNested => !Nested.IsDefault;
+            public bool IsNested => Nested != null;
```

Comment:
My reaction to `IsNested` is to think "is this variable nested".  In this case it seems to mean that it's a variable that has nested declarations.  I would assume this variable `HasNestedDeclarations` and those  variables would be `IsNested`.  

Am I missing something here?